### PR TITLE
Moving: The selector should not be able to jump between rows.

### PR DESCRIPTION
**Pull request by saralil**
Movement is now restricted so you can only move to a new position if there is a path to it. You can no longer jump between rows.

### CODE REVIEW

File: game_board.py [hunk 1]
```diff
@@ -148,60 +148,120 @@ def which_color_pair(stone_marker):
         color = 3
     return color
 
-def move_down(plus_list,current_row):
-    """ Finds the "+" char that is below the current_row. If the current "+" char is at the bottom, 
-        it finds the "+" char at the top with the same x position
-    Returns the new y position of the "+" 
+def move_down(plus_list,loc):
```
Comment (dakaza98) [inline game_board.py]: `loc` refers to the current location right? I would suggest renaming it to `current_location`

File: game_board.py [hunk 2]
```diff
@@ -148,60 +148,120 @@ def which_color_pair(stone_marker):
...
+    """ Finds the "+" char that is above loc. If the current "+" char is at a coordinate that
+        that violates the rules for movement, the positiono wont change.
```
Comment (dakaza98) [inline game_board.py]:
```suggestion
        that violates the rules for movement, the position wont change.
```

File: game_board.py [hunk 3]
```diff
@@ -148,60 +148,120 @@ def which_color_pair(stone_marker):
...
     """
-    
-    new_row = current_row
+    new_loc = loc
     
     while True:
-        new_row += 1
-        if new_row == len(plus_list):
-            new_row = 0
+        new_loc += 1
```
Comment (dakaza98) [inline game_board.py]: It's better if you write out the full name of the variable. It makes the code more readable
```suggestion
        new_location += 1
```

File: game_board.py [hunk 4]
```diff
@@ -148,60 +148,120 @@ def which_color_pair(stone_marker):
...
+        new_x = plus_list[new_loc][1]
 
-        current_y = plus_list[current_row][2]
-        new_y = plus_list[new_row][2]
+        current_y = plus_list[loc][2]
+        new_y = plus_list[new_loc][2]
         if current_x == new_x and current_y != new_y:
-          return new_row 
+            #Check if allowed to move down
+            if (int(current_x) != 16 and int(current_y) <= 6) or (int(current_x) == 16 and int(current_y) != 4 and int(current_y) != 12):
```
Comment (dakaza98) [inline game_board.py]: What is so special about the numbers 16, 4 and 12? Why do we need to check against these specific values. I would suggest putting them into variables that explain what they are or perhaps they can be calculated?

File: game_board.py [hunk 5]
```diff
@@ -363,13 +423,13 @@ def main(screen,player1_name,player2_name):
 
         print_choice(screen,current_row,plus_list,player1_turn)
         
-        screen.refresh()    
-
+        screen.refresh()   
+        #print(plus_list)
```
Comment (dakaza98) [inline game_board.py]: Remove this old print

File: game_board.py [hunk 6]
```diff
@@ -363,13 +423,13 @@ def main(screen,player1_name,player2_name):
...
         key = screen.getch()
-        if key == curses.KEY_LEFT and current_row > 0:  
-            current_row -= 1
-        elif key == curses.KEY_RIGHT and current_row < len(plus_list)-1:
-            current_row += 1
+        if key == curses.KEY_LEFT:  
+            current_row = move_left(plus_list, current_row)
+        elif key == curses.KEY_RIGHT: # and current_row < len(plus_list)-1:
```
Comment (dakaza98) [inline game_board.py]: Is that comment needed or can it be removed?

File: game_board.py [hunk 7]
```diff
@@ -148,60 +148,120 @@ def which_color_pair(stone_marker):
...
+        new_y = plus_list[new_loc][2]
+        if current_y == new_y and current_x != new_x:
+            #Check if allowed to move left
+            if (int(current_y) != 6 and int(current_x) <= 16) or (int(current_y) == 6 and int(current_x) != 12 and int(current_x) != 28):
+                return new_loc
+            else:
+                return loc
 
 
+def move_left(plus_list, loc):
```
Comment (dakaza98) [inline game_board.py]: When i try to move left at any position, I get a index out of range error

**Review by dakaza98 — CHANGES_REQUESTED** [review]

File: game_board.py (hunk 7 above)
Comment (espen21) [reply game_board.py]: fixed ,it was small error at line 253  new_x = plus_list[loc][1]  should be new_x = plus_list[new_loc][1]

File: game_board.py (hunk 4 above)
Comment (saralil) [reply game_board.py]: I made two new variables 'middle' and 'step' that represents the middle of the map and the length of a "step" along the x- and y-axis.

**Review by Antonfogelberg — APPROVED** [review]
:+1:

**Review by dakaza98 — APPROVED** [review]

**Pull request merged**